src/clr-angular/forms/common/wrapped-control.ts

```diff
@@ -110,21 +111,39 @@ export class WrappedFormControl<W extends DynamicWrapper> implements OnInit, OnD
   ngOnInit() {
     this._containerInjector = new HostWrapper(this.wrapperType, this.vcr, this.index);
     this.controlIdService = this._containerInjector.get(ControlIdService);
+
     if (this._id) {
       this.controlIdService.id = this._id;
     } else {
       this._id = this.controlIdService.id;
     }
-    if (this.renderer && this.el) {
-      this.renderer.setAttribute(this.el.nativeElement, 'aria-describedby', this.controlIdService.id + '-helper');
-    }
 
     if (this.ngControlService) {
       this.ngControlService.setControl(this.ngControl);
     }
+
+    this.listenForErrorStateChanges();
   }
 
   ngOnDestroy() {
     this.subscriptions.forEach(sub => sub.unsubscribe());
   }
+
+  private listenForErrorStateChanges() {
+    if (this.ifErrorService) {
+      this.subscriptions.push(
+        this.ifErrorService.statusChanges
+          .pipe(startWith(false), filter(() => this.renderer && !!this.el), distinctUntilChanged())
+          .subscribe(error => this.setAriaDescribedBy(error))
+      );
+    }
+  }
+
+  private setAriaDescribedBy(error: boolean) {
+    if (error) {
+      this.renderer.setAttribute(this.el.nativeElement, 'aria-describedby', this.controlIdService.id + '-error');
+    } else {
+      this.renderer.setAttribute(this.el.nativeElement, 'aria-describedby', this.controlIdService.id + '-helper');
+    }
```

Comment:
```typescript
this.renderer.setAttribute(
  this.el.nativeElement, 
  'aria-describedby', 
  this.controlIdService.id + ( error ? '-error' : '-helper' )
);
```

When I was reading the code a spend a lot of time comparing these two lines, character by character, just to find out that the difference is just the suffix that will be added.

Reply:
I think it would be better to separate this out even more so that the private method only has one job. Then the ternary could be either another private method or just a function...

```
private setAriaDescribedBy(error: boolean) {
      this.renderer.setAttribute(this.el.nativeElement, 'aria-describedby', this.getAriaDescribedById(error));
}

private getAriaDescribedById(error: boolean): string {
    return this.controlIdService.id.concat(error ? '-error' : '-helper' ));
}
```

This way we could test that the logic is working without having to test for its presence in the HTML.